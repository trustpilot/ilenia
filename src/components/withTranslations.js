import React from 'react';
import PropTypes from 'prop-types';

const withTranslations = (WrappedComponent) => {
  const contextConsumer = (props, context) =>
    <WrappedComponent {...props} translations={context.translations} locale={context.locale}/>;

  contextConsumer.contextTypes = {
    locale: PropTypes.string.isRequired,
    translations: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  };

  return contextConsumer;
};

export default withTranslations;
