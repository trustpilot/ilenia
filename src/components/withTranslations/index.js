import React from 'react';
import PropTypes from 'prop-types';

const withTranslations = (Component) => {
  const wrapper = (props, context) =>
    <Component {...props} translations={context.translations} locale={context.locale}/>;

  wrapper.contextTypes = {
    locale: PropTypes.string.isRequired,
    translations: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  };

  wrapper.displayName = `withTranslations(${Component.displayName || Component.name})`;

  return wrapper;
};

export default withTranslations;
