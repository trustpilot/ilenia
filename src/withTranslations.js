import React from 'react';

import LocalizationContext from './Context';

const withTranslations = (Component) => (props) => (
  <LocalizationContext.Consumer>
    {(context) => <Component {...props} translations={context.translations} locale={context.locale} />}
  </LocalizationContext.Consumer>
);

export default withTranslations;
