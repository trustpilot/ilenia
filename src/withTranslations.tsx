import * as React from 'react';

import { LocalizationContext } from '.';

function getDisplayName(WrappedComponent: React.ComponentType) {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
}

export const withTranslations = (Component: React.ComponentType) => {
  const WithTranslations = (props: any) => (
    <LocalizationContext.Consumer>
      {(context) => (
        <Component {...props} translations={context.translations} locale={context.locale} />
      )}
    </LocalizationContext.Consumer>
  );

  WithTranslations.displayName = `WithTranslations(${getDisplayName(Component)})`;
  WithTranslations.propTypes = Component.propTypes;
  WithTranslations.defaultProps = Component.defaultProps;
  return WithTranslations;
};
