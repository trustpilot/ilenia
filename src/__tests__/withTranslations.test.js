import React from 'react';
import 'raf/polyfill';
import { render } from 'react-testing-library';

import withTranslations from '../withTranslations';
import LocalizationProvider from '../LocalizationProvider';

const locale = 'da-DK';
const translations = {
  myString: 'Dansk',
};
const fallbackTranslations = {
  myString: 'English (US)',
};

const Child = (props) => (
  <div>
    <span>{JSON.stringify(props.translations)}</span>
    <span>{props.locale}</span>
  </div>
);

test('Wrapped component has access to locale and translations as props', () => {
  const TranslatedChild = withTranslations(Child);

  const { container } = render(
    <LocalizationProvider
      locale={locale}
      translations={{ ...fallbackTranslations, ...translations }}
    >
      <TranslatedChild />
    </LocalizationProvider>
  );

  expect(container.firstChild).toMatchSnapshot();
});
