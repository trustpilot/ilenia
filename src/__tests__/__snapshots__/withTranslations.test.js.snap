// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapped component has access to locale and translations as props 1`] = `
<div>
  <span>
    {"myString":"Dansk"}
  </span>
  <span>
    da-DK
  </span>
</div>
`;
